### ISSUE TITLE
[4.0] autoload_psr4.php not updating during extension install

### ISSUE THREAD

**Issue by joeforjoomla** [joomla/joomla-cms#35545]
It happened to a client of mine that installing a namespaced extension, the file autoload_psr4.php did not update including the namespacing of the newly installed extension.

To update the autoload_psr4.php including namespaces of the installed extension, it was needed to delete it and allow Joomla to create it again from scratch.

Obviously an extension doesn't work if its namespaces in autoload_psr4.php are not added.

Has someone any idea about how this could happen?

**Comment by Fedik** [comment]
It may happen when plugin `extension/namespacemap` is disabled

**Comment by joeforjoomla** [comment]
@Fedik you got the spot!

I've logged in on my client's website and indeed i've found the plugin disabled.

Would not be a good idea to prevent this to happen? Warn the user when unpublishing the plugin or lock it?

![image](https://user-images.githubusercontent.com/17835460/133063375-f1abcecf-14de-40ef-a780-f6bba34b6e2b.png)

**Comment by brianteeman** [comment]
iirc I raised it before that this is a plugin that should not be a plugin and that it should not be possible to disable it

**Comment by joeforjoomla** [comment]
@brianteeman you are totally right. Less than 1 month that J4 is out and problems with this issue are already started. I hope that this will be changed in 4.0.4

**Comment by brianteeman** [comment]
> I hope that this will be changed in 4.0.4

If you know how to then a PR would be most welcome.

**Comment by joeforjoomla** [comment]
I'm available to submit a PR about this matter. 
@brianteeman can you say me what do you prefer as a solution? Preventing that plugin to be disabled or inform the user with a message about potential problems if disabled?

**Comment by brianteeman** [comment]
Not sure I have a preference myself. Just one thing to be aware of is that if you are going down the message route (as done with the authentication plugins) then you dont see the message if you disable the plugin from the plugin manager list and you only see it if you open the plugin and disable it there.

Should it even be a plugin?

**Comment by joeforjoomla** [comment]
You right. Probably it was better if it wasn't designed as a plugin.

**Comment by joeforjoomla** [comment]
Perhaps something like the 'lock' icon for core extensions.
![image](https://user-images.githubusercontent.com/17835460/133081600-43c3f8e4-613f-402a-a3f9-570294116c92.png)

**Comment by brianteeman** [comment]
> Perhaps something like the 'lock' icon for core extensions.

That;s only in the extension manager to prevent it being uninstalled. It doesn't prevent it being disabled
